cfgov/v1/jinja2/v1/layouts/base.html

```diff
@@ -65,7 +65,7 @@
     <base target="_blank">
     {% endif %}
 
-    <link rel="canonical" href="{{ request.build_absolute_uri() | lower }}">
+    <link rel="canonical" href="{{ request.build_absolute_uri().split('?')[0] | lower }}">
```

Comment:
Is there ever a distinction here between `build_absolute_uri()` and something like `{{ request.scheme }}://{{ request.get_host }}{{ request.get_full_path }}`. It looks like probably not [from the `build_absolute_uri` source](https://github.com/django/django/blob/6a37e9bfae42eb180a8c978a3ade5b41c97c26b1/django/http/request.py#L204-L241).

My preference would to do less Python in templates than more, so either using the constituent properties of `request` or moving the string slicing into a template tag.

Reply:
Done in 5bfe0a1. I hard-coded https because that will be true everyone that canonical links matter

Reply:
From my testing, I think the `| lower` is superfluous as well (as there is a redirect or `lower` upstream), but I didn't look too hard at it

Reply:
I vaguely recall a need to explicitly lower-case some URLs for canonicalization purposes.

Reply:
Yeah, happy to leave it without a strong reason to remove

Reply:
Could we use `request.build_absolute_uri(request.path)` to avoid the need to strip the querystring in such an ugly way? See [docs](https://docs.djangoproject.com/en/4.2/ref/request-response/#django.http.HttpRequest.build_absolute_uri).